src/svc/crd/metabase.rs

```diff
@@ -0,0 +1,610 @@
+                    "Create event for custom resource",
+                );
+
+                recorder::normal(kube.to_owned(), &modified, action, message).await?;
+            }
+
+            // Stop reconciliation here and wait for next iteration, already
+            // triggered by the above patch request
+            return Ok(());
+        }
+
+        // ---------------------------------------------------------------------
+        // Step 3: upsert addon
+
+        info!(
+            kind = &kind,
+            namespace = &namespace,
+            name = &name,
+            "Upsert addon for custom resource",
+        );
+
+        let addon = modified.upsert(&apis).await?;
+
+        modified.set_addon_id(Some(addon.id.to_owned()));
+
+        debug!(
+            kind = &kind,
+            namespace = &namespace,
+            name = &name,
+            "Update information and status of custom resource",
+        );
+
+        let patch = resource::diff(&*origin, &modified).map_err(ReconcilerError::Diff)?;
+        let mut modified = resource::patch(kube.to_owned(), &modified, patch.to_owned())
+            .and_then(|modified| resource::patch_status(kube.to_owned(), modified, patch))
+            .await?;
+
+        let action = &Action::UpsertAddon;
+        let message = &format!(
+            "Create managed postgresql instance on clever-cloud '{}'",
+            addon.id
+        );
+        recorder::normal(kube.to_owned(), &modified, action, message).await?;
+
+        // ---------------------------------------------------------------------
+        // Step 4: upsert secret
+
+        let secrets = modified.secrets(&apis).await?;
+
+        // capture the url to update the status un Step 5
+        // this is needed because `Secret` takes ownership of the secrets
+        // alternatively we can just clone the whole `BTreeMap`
+        let url = secrets
+            .as_ref()
+            .and_then(|secrets| secrets.get("url").cloned());
+
+        if let Some(secrets) = secrets {
+            let s = secret::new(&modified, secrets);
+            let (s_ns, s_name) = resource::namespaced_name(&s);
+
+            info!(
+                kind = &kind,
+                namespace = &namespace,
+                name = &name,
+                "Upsert kubernetes secret resource for custom resource",
+            );
+
+            info!(
+                namespace = &s_ns,
+                name = &s_name,
+                "Upsert kubernetes secret",
+            );
+
+            let secret = resource::upsert(kube.to_owned(), &s, false).await?;
+
+            let action = &Action::UpsertSecret;
+            let message = &format!("Create kubernetes secret '{}'", secret.name_any());
+            recorder::normal(kube.to_owned(), &modified, action, message).await?;
+        }
+
+        // ---------------------------------------------------------------------
+        // Step 5: update the status
+
+        if let Some(url) = url {
```

Comment:
we should check that url is different from the current one to avoid unnecessary patch